tormenta-core/src/test/scala/com/twitter/tormenta/scheme/SchemeLaws.scala

```diff
@@ -19,16 +19,32 @@ package com.twitter.tormenta.scheme
 import org.scalatest._
 import scala.collection.JavaConverters._
 
-object SchemeWithHandlerSpecification extends WordSpec with Matchers {
-  "Scheme should handle failure" should {
-    "test failure" in {
-      val f: Array[Byte] => List[String] = b => throw new IllegalArgumentException("decode failed")
-
-      val schemeWithErrorHandler = Scheme(f).withHandler(t => List(t.getMessage))
-      val result = schemeWithErrorHandler.deserialize("test string".getBytes("UTF-8"))
+class SchemeWithHandlerSpecification extends WordSpec with Matchers {
+  "Scheme" should {
+    val f: Array[Byte] => List[String] = b => throw new IllegalArgumentException("decode failed")
 
+    def checkResult[T](scheme: Scheme[T], expectedResult: T) {
+      val result = scheme.deserialize("test string".getBytes("UTF-8"))
       assert(result.asScala.isEmpty == false) //test fails, returns an empty list
-      assert(result.asScala.head.get(0) == "decode failed")
+      assert(result.asScala.toList.map(_.get(0)) == expectedResult)
```

Comment:
I can't see that the return type is correct and this is not going to do something wonky.

Could you do: `(result.asScala.toList.map(_.get(0)): T) == expectedResult` so the compiler will fail if it is broken, or the new state of the art:
```scala
def eqv[T, U](t: T, u: U)(implicit ev: U =:= T): Boolean = (t == u)
```
then use eqv. Our old `def eqv[T](a: T, b: T) = a == b` doesn't always work because `T = Any` can always be inferred.